tests/test_attribute_factory_routes.py

```diff
@@ -138,3 +141,24 @@ def test_fully_failed_get_instances_handled_gracefully(
     assert response["nodes_response_status"] == "fail"
     assert len(response["errors"]) == 2
     assert response["responses"] == {"nb:Assessment": []}
+
+
+@pytest.mark.parametrize(
+    "path",
+    [
+        "assessments",
+        "diagnoses",
+        "pipelines",
+        "query",
+    ],
+)
+def test_node_request_urls_do_not_have_trailing_slash(path):
+    """
+    Ensure that URLs used to forward requests to node APIs do not include a trailing slash.
+    """
+    assert crud.build_node_request_urls(
+        ["https://node1.institute.org/", "https://node2.institute.org/"], path
+    ) == [
+        f"https://node1.institute.org/{path}",
+        f"https://node2.institute.org/{path}",
+    ]
```

Comment:
:cherries: but still important: let's keep the act and assert steps separate! i.e. build the nodes and store, then assert over the outcome. keeps it more readable and having a standard structure also helps